### ISSUE TITLE
windowsToWslSync returns incorrect path after second call

### ISSUE THREAD

**Issue by jmanuel1** [mojadev/wsl-path#11]
On my system, I have `C:\` mounted at `/mnt/c` in WSL. Let's say I pass the path `C:\Users\jamai\Documents\idris\tcs4\src\SmallStep.idr` to `windowsToWslSync`. Then I get the result I expected: `/mnt/c/Users/jamai/Documents/idris/tcs4/src/SmallStep.idr`. But then when the function is called again with the same path and cache, `windowsToWslSync` returns `/mnt/c/Users/jamai/Documents/idris/tcs4/src/Users/jamai/Documents/idris/tcs4/src/SmallStep.idr`. This is the resolution cache after the calls:

```json
{
	"basePathCache": {"wsl:C:\\": "/mnt/c/Users/jamai/Documents/idris/tcs4/src", "wsl:/mnt/c/Users/jamai/Documents/idris/tcs4/src": "C:\\"},
	"mountPoints": [],
	"wslCommand": "wsl"
}
```

I'm not sure how the resolution cache is supposed to work, but I'd think it should associate "C:\" with "/mnt/c".

If I always pass a new empty cache to `windowsToWslSync`, the function works.

**Comment by mojadev** [comment]
Hi, sorry for the long delay - I'll take a look!

**Comment by mojadev** [comment]
I can reproduce it, will fix it probably today - thanks for the detailed bug report

**Comment by mojadev** [comment]
I think the issue is related to the fact that the tool was written originally with WSL1, and probably the mount points need a different approach with WSL2. I'm right now working on a mac, but I should be able to test it later this day - worst case I'll need to ask a colleague this week.

**Comment by mojadev** [comment]
I published a version 3.0.1 that hopefully fixes the cause of this error.

The background is that this library doesn't spawn a process to resolve every path you use, but determines mount points first and resolves paths using those (see it as a pre-warmed cache containing the base paths you'll most likely use). Only in cases where this is not possible the wsl runtime is called to resolve paths, and results are cached afterwards.

For C:\\ this should never happen, but happened as the filesystem in WSL2 changed to 9p from drvfs, which had the effect that the mount points could not be determined anymore.

**Comment by jmanuel1** [comment]
I upgraded and things seem to be working as expected. Thank you!
